### DISCUSSION TITLE
Conditional rendering of nested template

### DISCUSSION THREAD

**Discussion by paulhaem** [alpinejs/alpine#3253] [7. Bugs (archived)] [answered]
Hey, I was using a nested template to conditonally render content based on an object property when I observed this odd behaviour. I was able to break down the scenario to this simple pen: https://codepen.io/paulhaem/pen/OJZwWpR

```html
<div
  x-data="{
    selectedField: null,
    setField() {
      this.selectedField = {
        title: 'Awesome!',
        subTitle: 'This is a field ...',
      }
    },
    clearField() {
      this.selectedField = null;
    }
  }"
>
  <button type="button" @click="setField">Set field</button>
  <button type="button" @click="clearField">Clear field</button>
  <template x-if="selectedField">
    <div>
      <h1 x-text="selectedField.title"></h1>
      <template x-if="selectedField.subTitle">
        <p x-text="selectedField.subTitle"></p>
      </template>
    </div>
  </template>
</div>
```

After using "set field" and "clear field" the second time following error and warning are being logged multiple times each time `clearField` is called:
```js
Uncaught TypeError: can't access property "subTitle", selectedField is null
```
```js
Alpine Expression Error: can't access property "subTitle", selectedField is null
```

The nested template tries to access the `subTitle` property of an empty object but should not be rendered in the first place as the parent template should only render if the object is set. Odd is that the functionality is not affected by these errors so it seems like some kind of code race.

**Comment by ekwoka** [comment, marked as answer]
It's basically just that the effects that manage the reactivity are queued when the value changes, and aren't necessarily cleared up in the process.

**Reply by paulhaem** [reply]
Is there anything to avoid those errors as they would eventually be interpreted as not-captured-errors when using Sentry for error tracking?

**Reply by SimoTod** [reply]
`<template x-if="selectedField?.subTitle">`?
